pkg/endpoint/bpf.go

```diff
@@ -342,16 +342,19 @@ func (e *Endpoint) hashHeaderfile(prefix string) (string, error) {
 	}
 	defer file.Close()
 
-	scanner := bufio.NewScanner(file)
 	hashWriter := md5.New()
-	for scanner.Scan() {
-		line := scanner.Text()
+	reader := bufio.NewReader(file)
+	for {
+		line, err := reader.ReadString('\n')
 		if strings.HasPrefix(line, "#") {
 			io.WriteString(hashWriter, line)
 		}
-	}
-	if err := scanner.Err(); err != nil {
-		return "", err
+		if err != nil {
```

Comment:
Shouldn't you check error first before checking if line `HasPrefix`?

Reply:
No, not with `iobuf.Reader`. In case the last line doesn't end with the delimiter, it returns both the line and `io.EOF`. This is the normal pattern with this type.

Reply:
In case of a real error, the line will be `""`, and I'll ignore that with `HasPrefix`. So it's all fine.